### PR TITLE
Merge `*QueryField` and `*QueryPath`

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

The difference between `Field` and `Path` is, that `Field` also is aware of `schema` and `version_id`. As the rest is common, this PR removes `Field` and adds `schema` and `version_id` to `Path`, but don't allow deserializing these variants. 

## 🔗 Related links

- [Asana task](https://app.asana.com/0/0/1203225514907853/f) _(internal)_

## 🔍 What does this change?

- Add `Schema` and `VersionId` to paths
- Remove `Field` and derived structures

## 📜 Does this require a change to the docs?

This change is internal only

## ⚠️ Known issues

`Schema` and `VersionId` are exposed outside of the `postgres` module, but the deduplication is very big.

## 🛡 What tests cover this?

A few tests were added to ensure, that `schema` and `version_id` are not exposed

### CODE REVIEW

File: packages/graph/hash_graph/lib/graph/src/ontology/entity_type.rs [hunk 1]
```diff
@@ -79,8 +81,8 @@ pub struct EntityTypeQueryPathVisitor {
 impl EntityTypeQueryPathVisitor {
     pub const EXPECTING: &'static str = "one of `ownedById`, `createdById`, `updatedById`, \
                                          `removedById`, `baseUri`, `versionedUri`, `version`, \
-                                         `title, `description`, `default`, `examples`, \
-                                         `properties`, `required`, `links`, or `requiredLinks`";
+                                         `title`, `description`, `default`, `examples`, \
+                                         `properties`, `required`, `links`, `requiredLinks`";
```
Comment (thehabbos007) [inline packages/graph/hash_graph/lib/graph/src/ontology/entity_type.rs:85]: The removal of the `or` at the end is intentional, right? Is it just to make it easier to add/change or is there another reason?

File: packages/graph/hash_graph/lib/graph/src/store/postgres/query/compile.rs [hunk 2]
```diff
@@ -46,11 +46,11 @@ impl<'f: 'q, 'q, T: PostgresQueryRecord<'q>> SelectCompiler<'f, 'q, T> {
...
     /// Creates a new compiler, which will default to select the fields returned from
-    /// [`PostgresQueryRecord::default_fields()`].
-    pub fn with_default_fields() -> Self {
+    /// [`PostgresQueryRecord::default_selection_paths()`].
+    pub fn with_default_selection() -> Self {
         let mut default = Self::new();
-        for field in T::default_fields() {
-            default.add_field(field);
+        for field in T::default_selection_paths() {
+            default.add_selection_path(field);
```
Comment (thehabbos007) [inline packages/graph/hash_graph/lib/graph/src/store/postgres/query/compile.rs:53]: nit:
```suggestion
        for path in T::default_selection_paths() {
            default.add_selection_path(path);
```

File: packages/graph/hash_graph/lib/graph/src/store/postgres/query/compile.rs [hunk 3]
```diff
@@ -46,11 +46,11 @@ impl<'f: 'q, 'q, T: PostgresQueryRecord<'q>> SelectCompiler<'f, 'q, T> {
     }
 
     /// Creates a new compiler, which will default to select the fields returned from
```
Comment (thehabbos007) [inline packages/graph/hash_graph/lib/graph/src/store/postgres/query/compile.rs:48]: Would we still call these fields here?

**Review by thehabbos007 — COMMENTED** [review]
Thanks, Tim. I have some comments which I think we should address before merging :)

File: packages/graph/hash_graph/lib/graph/src/store/postgres/query/compile.rs (hunk 3 above)
Comment (TimDiekmann) [reply packages/graph/hash_graph/lib/graph/src/store/postgres/query/compile.rs:48]: Actually, I think I will remove this method entirely later on, but I have not decided yet. This implies that requesting from a row is closer to the specified selection.

File: packages/graph/hash_graph/lib/graph/src/store/postgres/query/data_type.rs [hunk 4]
```diff
@@ -3,52 +3,35 @@ use type_system::DataType;
...
-
     fn base_table() -> Table {
         Table {
             name: TableName::DataTypes,
             alias: None,
         }
     }
 
-    fn default_fields() -> &'q [Self::Field] {
+    fn default_selection_paths() -> &'q [Self::Path<'q>] {
```
[the remark this answers is not in the document]
Comment (TimDiekmann) [reply packages/graph/hash_graph/lib/graph/src/store/postgres/query/data_type.rs:17]: `VersionedUri` corresponds to `"schema"->>'$id'`. `Schema` will only return the JSON blob.

File: packages/graph/hash_graph/lib/graph/src/store/postgres/query/compile.rs (hunk 3 above)
Comment (thehabbos007) [reply packages/graph/hash_graph/lib/graph/src/store/postgres/query/compile.rs]: Could you say more about what "requesting from a row is closer to the specified selection" means?

File: packages/graph/hash_graph/lib/graph/src/store/postgres/query/data_type.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (thehabbos007) [reply packages/graph/hash_graph/lib/graph/src/store/postgres/query/data_type.rs:17]: I see. That makes sense

File: packages/graph/hash_graph/lib/graph/src/store/postgres/query/compile.rs (hunk 3 above)
Comment (TimDiekmann) [reply packages/graph/hash_graph/lib/graph/src/store/postgres/query/compile.rs]: #1258 currently uses this method to create a compiler with the default selection. A few lines after creating the compiler it assumes that the returned `Row` has some specific columns.

With this PR it's possible to always use the default compiler and manually adding the `Path`, which then can be returned by `row.get(x)`.

**Review by thehabbos007 — APPROVED** [review]

**Pull request merged**